### PR TITLE
Implemented IexHistoricalPrices endpoint.

### PR DESCRIPTION
**Pull request by samallenswe**
IexHistoricalCap

### CODE REVIEW

**Review by grundygolem2 — COMMENTED** [review]
This looks good. Could you add another test that tests for non-date queries? i.e. the 5d query?

**Comment by samallenswe** [comment]
> This looks good. Could you add another test that tests for non-date queries? i.e. the 5d query?

Okay. I'll get that added

File: src/main/java/org/galatea/starter/service/IexClient.java [hunk 1]
```diff
@@ -32,4 +34,7 @@
   @GetMapping("/tops/last")
   List<IexLastTradedPrice> getLastTradedPriceForSymbols(@RequestParam("symbols") String[] symbols);
 
+  @GetMapping("/stable/stock/{symbol}/chart/{range}/{date}?chartByDay=true")
```
Comment (jbrody17) [inline src/main/java/org/galatea/starter/service/IexClient.java:37]: Why do we want the chart by day = true for all requests

File: src/main/java/org/galatea/starter/service/IexClient.java [hunk 2]
```diff
@@ -32,4 +34,7 @@
...
+  List<IexHistoricalPrices> getHistoricalPrices(@PathVariable("symbol") String symbol, @PathVariable("range") String dateRange, @PathVariable("date") String date);
```
Comment (jbrody17) [inline src/main/java/org/galatea/starter/service/IexClient.java:38]: Are these all required parameters?

File: src/main/java/org/galatea/starter/service/IexService.java [hunk 3]
```diff
@@ -45,5 +46,14 @@
...
-
+  /**
+   * Get the historical prices for each Symbol and date range that is passed in.
+   *
+   * @param symbol the list of symbols to get a last traded price for.
+   * @param dateRange the range of dates to search for historical prices.
+   * @return a list of historical prices objects for each Symbol that is passed in.
+   */
+  public List<IexHistoricalPrices> getHistoricalPrices(final String symbol, final String dateRange, final String date) {
+    return iexClient.getHistoricalPrices(symbol,dateRange,date);
```
Comment (jbrody17) [inline src/main/java/org/galatea/starter/service/IexService.java]: Better style to have spacing between method parameters

**Review by jbrody17 — COMMENTED** [review]
Mostly looks good, couple of little things. One thing I didn't see was an IEX token. Have you tested this? Typically for API calls like this, the service requires a token of some sort

File: src/main/java/org/galatea/starter/service/IexService.java (hunk 3 above)
Comment (samallenswe) [reply src/main/java/org/galatea/starter/service/IexService.java]: Missed that, will fix.

File: src/main/java/org/galatea/starter/service/IexClient.java (hunk 1 above)
Comment (samallenswe) [reply src/main/java/org/galatea/starter/service/IexClient.java:37]: When making the request with a specific date in mind, adding this to the end of the query will return only the OHLCV chart information for that date.

File: src/main/java/org/galatea/starter/service/IexClient.java (hunk 2 above)
Comment (samallenswe) [reply src/main/java/org/galatea/starter/service/IexClient.java:38]: In my current implementation, yes. I'm handling the case for passing a value of "date" to the range parameter, doing so requires us to have a third parameter for the specific date.

As part of handling non-date queries for testing (see Casey's comment above), I'm going to add a default case for when we only use symbol and range parameters.

**Review by grundygolem2 — APPROVED** [review]
Looks good

**Comment by jbrody17** [comment]
Sweet! Like the changes

**Pull request merged**